### PR TITLE
Properly process random_state when fitting Time Series Forest ensemble in parallel

### PR DESCRIPTION
**Pull request by kachayev**
**What does this implement/fix? Explain your changes.**

`BaseTimeSeriesForest` handles random state non-deterministically when `n_jobs` is > 1. Now random state is properly initialized into each estimator in the ensemble before passing to parallel scheduler. 

As a follow up to #796. I assume, `_clone_estimator` helper should be moved to utils (I can do  this after #796 is merged).

@mloning Same question as with ROCKET PR: put additional if-else branch to keep reproducibility when `n_jobs=1` or just update test case?

### CODE REVIEW

**Comment by mloning** [comment]
Thanks @kachayev - my preference is the same here: fix the bug and update test case avoiding if-else logic. @TonyBagnall let me know if you disagree.

**Comment by kachayev** [comment]
@mloning I updated test case. Will do another update after #796 is merged.

File: sktime/series_as_features/base/estimators/interval_based/_tsf.py
Comment (mloning) [inline sktime/series_as_features/base/estimators/interval_based/_tsf.py:178]: Could we move this to utils here or in PR #796?

**Review by mloning — COMMENTED** [review]
Thanks @kachayev - looks mostly good to me. 

@TonyBagnall @MatthewMiddlehurst could you confirm that you're happy with the changes to the references values that we test against?

File: sktime/series_as_features/base/estimators/interval_based/_tsf.py
Comment (kachayev) [reply sktime/series_as_features/base/estimators/interval_based/_tsf.py:178]: I was thinking about doing this in #796 but now I don't know which of these PRs have a chance to be merged first (I need to update one of them, merge changes into a seconds one and, only after that, rework the code).

Comment (mloning) [reply sktime/series_as_features/base/estimators/interval_based/_tsf.py:178]: Alright makes sense, let's leave it as it is and we see which one gets merged first

Comment (mloning) [reply sktime/series_as_features/base/estimators/interval_based/_tsf.py:178]: @kachayev now that #796 is merged, I can merge this one once you move the function to utils. Thanks!

Comment (kachayev) [reply sktime/series_as_features/base/estimators/interval_based/_tsf.py:178]: @mloning on it!

Comment (mloning) [reply sktime/series_as_features/base/estimators/interval_based/_tsf.py:178]: @kachayev the failing linux builds on main are now fixed, happy to merge this one once the function is moved to utils and reused in the other code

**Comment by mloning** [comment]
@kachayev we added some more stringent docstring tests (#752) , let me know if you need help with those. We also still need to move the `_clone_estimator` function before we can merge this.

**Review by mloning — APPROVED** [review]
@kachayev I added the suggested changes to get this PR merged, hope that's okay for you. 

@MatthewMiddlehurst @TonyBagnall let me know if you want anything changed or have any other comments before we merge.

**Comment by kachayev** [comment]
@mloning Sure, thanks for helping to finish this one!

**Review by MatthewMiddlehurst — APPROVED** [review]
Havent really done much with the TSF code but I dont see any issues with the changes. I think Arsenal still has a clone_estimator method which could be replaced?

**Comment by mloning** [comment]
Thanks @kachayev and @MatthewMiddlehurst - now merged!

**Pull request merged**
